Remove redundancy in variable usage, variable qualifiers and unused variables

I'm trying to compile CXXGraph inside include/ for a private codebase and since the repo uses -Werror, -Wunused-variables and -Wignorequalifiers these errors happen.

I was wondering if theres any interest in trying to slim down these warnings directly in the library?

```
/Users/jjasmine/Developer/github/anon/include/CXXGraph/Edge/Edge_decl.h:60:3: error: 'const' type qualifier on return type has no effect [-Werror,-Wignored-qualifiers]
  const unsigned long long getId() const;
  ^~~~~~
In file included from /Users/jjasmine/Developer/github/anon/standalone/source/main.cpp:4:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/CXXGraph.hpp:7:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/Edge/Edge.h:25:
/Users/jjasmine/Developer/github/anon/include/CXXGraph/Edge/Edge_impl.hpp:80:1: error: 'const' type qualifier on return type has no effect [-Werror,-Wignored-qualifiers]
const unsigned long long Edge<T>::getId() const {
^~~~~~
In file included from /Users/jjasmine/Developer/github/anon/standalone/source/main.cpp:4:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/CXXGraph.hpp:11:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/Graph/Graph.h:23:
/Users/jjasmine/Developer/github/anon/include/CXXGraph/Graph/Graph_impl.hpp:460:8: error: unused variable 'undirected' [-Werror,-Wunused-variable]
  bool undirected = this->isUndirectedGraph();
       ^
In file included from /Users/jjasmine/Developer/github/anon/standalone/source/main.cpp:4:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/CXXGraph.hpp:11:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/Graph/Graph.h:31:
/Users/jjasmine/Developer/github/anon/include/CXXGraph/Graph/Algorithm/CycleDetection_impl.hpp:47:7: error: unused variable 'stateCounter' [-Werror,-Wunused-variable]
  int stateCounter = 0;
      ^
In file included from /Users/jjasmine/Developer/github/anon/standalone/source/main.cpp:4:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/CXXGraph.hpp:11:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/Graph/Graph.h:47:
/Users/jjasmine/Developer/github/anon/include/CXXGraph/Graph/IO/InputOperation_impl.hpp:32:63: error: unused parameter 'compress' [-Werror,-Wunused-parameter]
                           const std::string &OFileName, bool compress,
                                                              ^
In file included from /Users/jjasmine/Developer/github/anon/standalone/source/main.cpp:4:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/CXXGraph.hpp:11:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/Graph/Graph.h:48:
/Users/jjasmine/Developer/github/anon/include/CXXGraph/Graph/IO/OutputOperation_impl.hpp:33:62: error: unused parameter 'compress' [-Werror,-Wunused-parameter]
                          const std::string &OFileName, bool compress,
                                                             ^
In file included from /Users/jjasmine/Developer/github/anon/standalone/source/main.cpp:4:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/CXXGraph.hpp:16:
/Users/jjasmine/Developer/github/anon/include/CXXGraph/Partitioning/EdgeBalancedVertexCut.hpp:106:10: error: unused variable 'MAX_SCORE' [-Werror,-Wunused-variable]
  double MAX_SCORE = 0.0;
         ^
In file included from /Users/jjasmine/Developer/github/anon/standalone/source/main.cpp:4:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/CXXGraph.hpp:19:
/Users/jjasmine/Developer/github/anon/include/CXXGraph/Partitioning/Partition.hpp:277:16: error: unused variable 'numberOfEdges' [-Werror,-Wunused-variable]
  unsigned int numberOfEdges = 0;
               ^
/Users/jjasmine/Developer/github/anon/include/CXXGraph/Partitioning/Partition.hpp:292:16: error: unused variable 'numberOfNodes' [-Werror,-Wunused-variable]
  unsigned int numberOfNodes = 0;
               ^
In file included from /Users/jjasmine/Developer/github/anon/standalone/source/main.cpp:4:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/CXXGraph.hpp:23:
In file included from /Users/jjasmine/Developer/github/anon/include/CXXGraph/Partitioning/Partitioner.hpp:39:
/Users/jjasmine/Developer/github/anon/include/CXXGraph/Partitioning/WeightBalancedLibra.hpp:140:14: error: unused variable 's_node' [-Werror,-Wunused-variable]
      size_t s_node = (u_degree > v_degree) ? v : u;
             ^
/Users/jjasmine/Developer/github/anon/include/CXXGraph/Partitioning/WeightBalancedLibra.hpp:141:14: error: unused variable 't_node' [-Werror,-Wunused-variable]
      size_t t_node = (u_degree > v_degree) ? u : v;
             ^
11 errors generated.
make[2]: *** [standalone/CMakeFiles/anonStandalone.dir/source/main.cpp.o] Error 1
make[1]: *** [standalone/CMakeFiles/anonStandalone.dir/all] Error 2
```

--- a/include/CXXGraph/Edge/Edge_decl.h
+++ b/include/CXXGraph/Edge/Edge_decl.h
@@ -59,7 +59,7 @@ class Edge {
   virtual ~Edge() = default;
   void setFirstNode(shared<const Node<T>> node);
   void setSecondNode(shared<const Node<T>> node);
-  const unsigned long long getId() const;
+  unsigned long long getId() const;
   const std::pair<shared<const Node<T>>, shared<const Node<T>>> &getNodePair()
       const;
   shared<const Node<T>> getOtherNode(shared<const Node<T>> node) const;

--- a/include/CXXGraph/Edge/Edge_impl.hpp
+++ b/include/CXXGraph/Edge/Edge_impl.hpp
@@ -79,7 +79,7 @@ void Edge<T>::setSecondNode(shared<const Node<T>> node) {
 }
 
 template <typename T>
-const unsigned long long Edge<T>::getId() const {
+unsigned long long Edge<T>::getId() const {
   return id;
 }
 

--- a/include/CXXGraph/Graph/Algorithm/CycleDetection_impl.hpp
+++ b/include/CXXGraph/Graph/Algorithm/CycleDetection_impl.hpp
@@ -44,7 +44,6 @@ bool Graph<T>::isCyclicDirectedGraphDFS() const {
   for (const auto &node : nodeSet) {
     state[node->getId()] = not_visited;
   }
-  int stateCounter = 0;
 
   // Start visiting each node.
   for (const auto &node : nodeSet) {

--- a/include/CXXGraph/Partitioning/EdgeBalancedVertexCut.hpp
+++ b/include/CXXGraph/Partitioning/EdgeBalancedVertexCut.hpp
@@ -104,7 +104,6 @@ void EdgeBalancedVertexCut<T>::performStep(shared<const Edge<T>> e,
   //*** Check which partition has the less load
   int MIN_LOAD = state->getMachineLoad(0);
   int machine_id = 0;
-  double MAX_SCORE = 0.0;
 
   for (int m = 0; m < P; m++) {
     int load = state->getMachineLoad(m);

--- a/include/CXXGraph/Partitioning/Partition.hpp
+++ b/include/CXXGraph/Partitioning/Partition.hpp
@@ -274,7 +274,6 @@ unsigned int getMinNodesLoad(const PartitionMap<T> &partitionMap) {
 
 template <typename T>
 unsigned int getNumberOfEdges(const PartitionMap<T> &partitionMap) {
-  unsigned int numberOfEdges = 0;
   T_EdgeSet<T> edgeSet;
 
   for (const auto &it : partitionMap) {
@@ -289,7 +288,6 @@ unsigned int getNumberOfEdges(const PartitionMap<T> &partitionMap) {
 
 template <typename T>
 unsigned int getNumberOfNodes(const PartitionMap<T> &partitionMap) {
-  unsigned int numberOfNodes = 0;
   std::unordered_set<shared<const Node<T>>, nodeHash<T>> nodeSet;
 
   for (const auto &it : partitionMap) {
@@ -358,8 +356,7 @@ std::ostream &operator<<(std::ostream &os, const Partition<T> &partition) {
       os << *std::static_pointer_cast<const UndirectedEdge<T>>(*it) << "\n";
     } else {
       // Should never happens
-      os << "Wrong Edge Class"
-         << "\n";
+      os << "Wrong Edge Class" << "\n";
     }
   }
   return os;

--- a/include/CXXGraph/Partitioning/WeightBalancedLibra.hpp
+++ b/include/CXXGraph/Partitioning/WeightBalancedLibra.hpp
@@ -138,8 +138,6 @@ void WeightBalancedLibra<T>::performStep(shared<const Edge<T>> e,
 
       // according to paper, s refers to node with lower degree, t = {u, v} -
       // {s}
-      size_t s_node = (u_degree > v_degree) ? v : u;
-      size_t t_node = (u_degree > v_degree) ? u : v;
       const std::set<int> &s_partition =
           (u_degree > v_degree) ? v_partition : u_partition;
       const std::set<int> &t_partition =

